met-web/src/components/layout/Header/InternalHeader.tsx

```diff
@@ -1,104 +1,424 @@
+const TenantSelector = ({
+    isVisible,
+    onStateChange,
+}: {
+    isVisible: boolean;
+    onStateChange?: (isOpen: boolean) => void;
+}) => {
+    const shouldUseMobileView = useMediaQuery((theme: Theme) => theme.breakpoints.down('md'));
+    const currentTenant = useAppSelector((state) => state.tenant);
+    const tenants = (useAsyncValue() as Tenant[]).filter((tenant) => tenant.short_name !== currentTenant.short_name);
+    const noOtherTenants = tenants.length === 0;
+    const tenantDropdownButton = useRef<HTMLButtonElement>(null);
+    const [tenantDrawerOpen, setTenantDrawerOpen] = useState(false);
+
+    useEffect(() => {
+        if (!isVisible) setTenantDrawerOpen(false);
+    }, [isVisible, setTenantDrawerOpen]);
+
+    useEffect(() => {
+        if (onStateChange) onStateChange(tenantDrawerOpen);
+    }, [tenantDrawerOpen, onStateChange]);
+
+    if (noOtherTenants) {
+        return (
+            <Link to="/home">
+                <BodyText sx={{ userSelect: 'none' }}>{currentTenant.name}</BodyText>
+            </Link>
+        );
+    }
+
+    const tenantList = tenants.map((tenant) => (
+        <MenuItem
+            key={tenant.short_name}
+            component={Link}
+            href={`/${tenant.short_name}/home`}
+            sx={{
+                paddingLeft: 2,
+                paddingRight: 2,
+                fontSize: '16px',
+                textTransform: 'capitalize',
+            }}
+        >
+            {tenant.name}
+        </MenuItem>
+    ));
+
+    if (shouldUseMobileView) {
+        return (
+            <>
+                <ButtonBase
+                    aria-haspopup
+                    aria-controls={tenantDrawerOpen ? 'tenant-drawer' : undefined}
+                    aria-expanded={tenantDrawerOpen}
+                    aria-label={'Tenant Switcher'}
+                    onClick={() => {
+                        setTenantDrawerOpen(!tenantDrawerOpen);
+                    }}
+                    sx={{
+                        height: 'calc( 100% - 2px )' /* 1px on either side to show border*/,
+                        ...dropdownMenuStyles,
+                    }}
+                >
+                    <TenantButtonContent isOpen={tenantDrawerOpen} />
+                </ButtonBase>
+                <Drawer
+                    anchor="top"
+                    open={tenantDrawerOpen}
+                    onClose={() => {
+                        setTenantDrawerOpen(false);
+                    }}
+                    sx={{
+                        zIndex: (theme: Theme) => theme.zIndex.drawer + 3, // render under app bar but above side nav
+                        '& .MuiDrawer-paper': {
+                            padding: '1rem',
+                            top: '6.5rem',
+                            backgroundImage: 'none',
+                            borderBottomRightRadius: '16px',
+                        },
+                    }}
+                >
+                    <MenuList
```

Comment:
I'd recommend we set this to be a `nav` element if possible. I think any time you use a menu to navigate across the site, it should have a `nav` tag and be aria labelled, ideally.